src/WebJobs.Script.WebHost/Health/TokenExpirationService.cs

```diff
@@ -43,19 +43,26 @@ public TokenExpirationService(IEnvironment environment, ILogger<TokenExpirationS
 
         protected override async Task ExecuteAsync(CancellationToken cancellationToken)
         {
-            await Task.Yield(); // force a yield, so our synchronous code does not slowdown startup
-            await _standby.Task.WaitAsync(cancellationToken);
+            try
+            { 
+                await Task.Yield(); // force a yield, so our synchronous code does not slowdown startup
+                await _standby.Task.WaitAsync(cancellationToken);
 
-            if (_environment.IsCoreTools())
-            {
-                return;
-            }
+                if (_environment.IsCoreTools())
+                {
+                    return;
+                }
 
-            while (!cancellationToken.IsCancellationRequested)
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    // check at every hour
+                    AnalyzeSasTokenInUri();
+                    await Task.Delay(TimeSpan.FromHours(1), cancellationToken);
+                }
+            }
+            catch (Exception e)
             {
-                // check at every hour
-                AnalyzeSasTokenInUri();
-                await Task.Delay(TimeSpan.FromHours(1), cancellationToken);
+                _logger.LogError(e, "Error occurred in {ServiceName}", nameof(TokenExpirationService));
```

Comment:
Question: So one failure will cause the `AnalyzeSasTokenInUri` to be never called again until the app restarts?. Is that by design? Or should we continue calling `AnalyzeSasTokenInUri();` every hour? (that means move the try-catch block inside the while

Reply:
I don't know if we want to run every hour if the service is broken. A fix for this service to work would be a config update which would lead to a restart anyway?

Side note: we should log warning not error

Reply:
Yeah we wouldn't want this token expiration service to be called anymore if something threw an error initially since there's some error being thrown likely because of the user's app configs not matching what we were expecting